newdle/client/src/components/answer/AnswerGrid.js

```diff
@@ -0,0 +1,703 @@
+import React, {useState} from 'react';
+import {useDispatch, useSelector} from 'react-redux';
+import {Trans} from '@lingui/macro';
+import _ from 'lodash';
+import moment from 'moment';
+import PropTypes from 'prop-types';
+import {Icon, Table, Radio, Popup} from 'semantic-ui-react';
+import {setAnswer} from '../../actions';
+import {getAnswers} from '../../answerSelectors';
+import {
+  getNewdleDuration,
+  getNewdleParticipants,
+  getNewdleTimezone,
+  getNewdleTimeslots,
+} from '../../selectors';
+import {serializeDate, toMoment} from '../../util/date';
+import AvailabilityRing from '../AvailabilityRing';
+import styles from './AnswerGrid.module.scss';
+
+function formatMeetingTime(startTime, duration) {
+  const endTime = moment(startTime)
+    .add(duration, 'm')
+    .format('HH:mm');
+  return `${serializeDate(startTime, 'HH:mm')} - ${endTime}`;
+}
+
+/**
+ * Computes the width and offset of the red strips used in the answer grid view,
+ * which denote the busy time duration. A single timeslot can have multiple of
+ * these strips as multiple busy times can overlap with a given timeslot.
+ *
+ * The red strip has absolute positioning with respect to the parent table cell and
+ * is styled using a combination of 'width' and 'left' or 'right' CSS rules.
+ *
+ */
+function getBusyTimesPositions(timeslot, busyTimes, duration) {
+  const start = toMoment(timeslot, moment.HTML5_FMT.DATETIME_LOCAL);
+  const styles = [];
+
+  Object.entries(busyTimes).forEach(([date, times]) => {
+    if (start.isSame(date, 'day')) {
```

Comment:
What about using something like `busyTimes[start]` instead of the loop? or `.find()` on the entries if you need the momentjs logic (but just converting `start` to a date string would probably work as well)

Reply:
Refactored to remove the loop :+1: